### PR TITLE
fix(skia): incorrect localization culture

### LINKED ISSUE #16353: Skia Localization is using incorrect culture
### Current behavior

As per these settings:

image

WinUI apps display in English, whereas GTK and WPF apps show up in Czech (for an app that has both `en` and `cs` strings.

### Expected behavior

Behavior should match

### How to reproduce it (as minimally and precisely as possible)

_No response_

### Workaround

_No response_

### Works on UWP/WinUI

None

### Environment

_No response_

### NuGet package version(s)

_No response_

### Affected platforms

_No response_

### IDE

_No response_

### IDE version

_No response_

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by Xiaoy312**
GitHub Issue (If applicable): closes #16353

## PR Type

What kind of change does this PR introduce?
- Bugfix

## What is the current behavior?
On skia-desktop, the default thread culture is set to Windows' "Regional format" culture.

## What is the new behavior?
Aligned uno (not just limited to skia-desktop) ResourceLoader behaviors more closely to Windows.

## PR Checklist
Please check if your PR fulfills the following requirements:
- [ ] ~~Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)~~
- [ ] ~~[Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)~~
- [ ] ~~Validated PR `Screenshots Compare Test Run` results.~~
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

**Comment by Xiaoy312** [comment]
fix failed:
1. none of the options within `GetManifestLanguages` should be considered
2. they are other factor(s) beside #-1 that also affects resource resolution (both: uno + uno.extensions) + cultures

File: src/Uno.UWP/Globalization/ApplicationLanguages.cs [hunk 1]
```diff
@@ -116,7 +116,9 @@ private static string[] GetManifestLanguages()
 #if __ANDROID__
 				ContextHelper.Current?.Resources?.Configuration?.Locales?.Get(0)?.ToLanguageTag(),
 #endif
-				CultureInfo.InstalledUICulture?.Name,
+				// On desktop-skia, InstalledUICulture returns the Windows' Regional format.
+				// We should not relies on this, as WinUI doesn't use this for UI other than date/time.
```
Comment (jeromelaban) [inline src/Uno.UWP/Globalization/ApplicationLanguages.cs:120]:
```suggestion
				// We should not rely on this, as WinUI doesn't use this for UI other than date/time.
```

File: src/Uno.UWP/ApplicationModel/Resources/ResourceLoader.cs [hunk 2]
```diff
@@ -14,340 +17,396 @@
...
-			// If we have a culture that doesn't specify country/region, we want to match its specific cultures if it's not found.
-			// For example, if we have es and it's not found, we want to match es-MX
-			if (culture.IsNeutralCulture)
+		context = _loaderContext;
+		return false;
+	}
+
+	private static string[] GetLanguagePreferences(LoaderContext context)
+	{
+		return (ApplicationLanguages.Languages ?? Array.Empty<string>())
```
Comment (Xiaoy312) [inline src/Uno.UWP/ApplicationModel/Resources/ResourceLoader.cs:167]: fixme: ensure we arent breaking existing impl for mobile & non-skia-desktop

File: src/Uno.UWP/FeatureConfiguration/WinRTFeatureConfiguration.cs [hunk 3]
```diff
@@ -19,4 +19,27 @@ public static class ApplicationLanguages
...
+		/// </summary>
+		public static bool PreserveParsedResources { get; set; }
+
+		// todo@xy: not impl
+		// fixme@xy: is this even needed?
+		/// <summary>
+		/// Determines if the ResourceLoader can fallback to the base or one of the sibling culture when a resource fails to resolve with the given languages/cultures.
+		/// Enables resolution between cultures: fr-CA <--> fr-FR, fr <--> fr-CA.
+		/// </summary>
+		public static bool AllowBaseOrSiblingCulturesFallback { get; set; } = true;
```
Comment (Xiaoy312) [inline src/Uno.UWP/FeatureConfiguration/WinRTFeatureConfiguration.cs:43]: we probably dont need such flag?
legacy uno behave as if this is true, so does windows

File: src/Uno.UWP/Globalization/ApplicationLanguages.cs
Comment (Xiaoy312) [inline src/Uno.UWP/Globalization/ApplicationLanguages.cs:1]: todo: review relationship/interaction between (GP.Languagues, AL.Languages\PLO\ManifestLangauges)
and their usage in ResourceLoader
across all targets...

Comment (Xiaoy312) [reply src/Uno.UWP/Globalization/ApplicationLanguages.cs:1]: currently some of the base properties throws on certain platforms for not-impl/supported
we should handle that more gracefully/elegantly? while not breaking the existing impl...

**Comment by agneszitte** [comment]
@jeromelaban, @Xiaoy312  CI build is green, additional changes needed or good for a last review?

File: src/Uno.UWP/System/UserProfile/GlobalizationPreferences.cs [hunk 4]
```diff
@@ -1,14 +1,57 @@
...
-	public static partial class GlobalizationPreferences
+
+#if __ANDROID__ || __IOS__ || __SKIA__
+	public static IReadOnlyList<string> Languages =>
+#if __ANDROID__
+		new[] { Locale.Default.ToLanguageTag() };
+#elif __IOS__
+		NSLocale.PreferredLanguages;
+#elif __SKIA__
+		OperatingSystem.IsWindows() ? GetWinUserLanguageList() : throw new PlatformNotSupportedException();
```
Comment (Youssef1313) [inline src/Uno.UWP/System/UserProfile/GlobalizationPreferences.cs:26]: > throw new PlatformNotSupportedException()

No way to have something on other platforms? Some fallback? Preserving old behavior?

Comment (Xiaoy312) [reply src/Uno.UWP/System/UserProfile/GlobalizationPreferences.cs:26]: there were no prior impl other than android + ios, which are both preserved

File: src/Uno.UWP/ApplicationModel/Resources/ResourceLoader.cs [hunk 5]
```diff
@@ -14,340 +17,381 @@
...
 
-	public ResourceLoader(string name) : this(name, true)
-	{
-	}
+	[NotImplemented]
+	public static string GetStringForReference(Uri uri) => throw new NotSupportedException();
+}
 
+partial class ResourceLoader // static
+{
```
Comment (jeromelaban) [inline src/Uno.UWP/ApplicationModel/Resources/ResourceLoader.cs:42]: Feels like this should be separate files.

File: src/Uno.UWP/ApplicationModel/Resources/ResourceLoader.cs [hunk 6]
```diff
@@ -14,340 +17,381 @@
...
-			ReloadResources(current, defaultLanguage);
+			return
+				c1 is { } &&
+				c2 is { } &&
+				GetBaseCulture(c1) == GetBaseCulture(c2);
 		}
+	}
+}
+
+partial class ResourceLoader // instanced
```
Comment (jeromelaban) [inline src/Uno.UWP/ApplicationModel/Resources/ResourceLoader.cs:290]: Same here.

File: src/Uno.UWP/System/UserProfile/GlobalizationPreferences.cs (hunk 4 above)
Comment (jeromelaban) [inline src/Uno.UWP/System/UserProfile/GlobalizationPreferences.cs:26]: It's probably best to return an empty list, or maybe return the default thread culture.

**Review by MartinZikmund — APPROVED** [review]

File: src/Uno.UWP/WinRT/MarshalString.skia.cs [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+	{
+		if (hstring != IntPtr.Zero)
+		{
+			_ = Platform.WindowsDeleteString(hstring);
+		}
+	}
+
+	private static class Platform
+	{
+		[DllImport("api-ms-win-core-winrt-string-l1-1-0.dll", CallingConvention = CallingConvention.StdCall)]
```
Comment (MartinZikmund) [inline src/Uno.UWP/WinRT/MarshalString.skia.cs:34]: Will this fail gracefully when running on older versions of Windows?

Comment (Xiaoy312) [reply src/Uno.UWP/WinRT/MarshalString.skia.cs:34]: this is likely to fail fatally unless you are on windows 10 or later
any recommendation on how to recover from?

Comment (Xiaoy312) [reply src/Uno.UWP/WinRT/MarshalString.skia.cs:34]: given the fact that we only support minimum of .net7 (soon 8) or higher now
and starting from net7, win7 and win8.1 are no longer supported
i think this is a non-issue

Comment (Xiaoy312) [reply src/Uno.UWP/WinRT/MarshalString.skia.cs:34]: @jeromelaban @MartinZikmund what are your opinions?

Comment (jeromelaban) [reply src/Uno.UWP/WinRT/MarshalString.skia.cs:34]: Windows 7, 8 and a few other Windows 10 are not supported anymore, and people are having increasingly difficult time to get it running there (https://github.com/dotnet/core/issues/7556#issuecomment-1786377918). 

If we find someone broken by this, we'll adjust if we can.

**Pull request merged**